pkg/apis/ibmcloudprovider/v1beta1/register.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package v1beta1 contains API Schema definitions for the ibmcloudprovider v1beta1 API group
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen=package,register
+// +k8s:conversion-gen=github.com/openshift/cluster-api-provider-ibmcloud/pkg/apis/ibmcloudprovider
+// +k8s:defaulter-gen=TypeMeta
+// +groupName=ibmcloudprovider.machine.openshift.io
+package v1beta1
+
+import (
+	"fmt"
+	"encoding/json"
+
+	yaml "sigs.k8s.io/yaml"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	klog "k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/scheme"
+)
+
+var (
+	// SchemeGroupVersion is group version used to register these objects
+	SchemeGroupVersion = schema.GroupVersion{Group: "ibmcloudprovider.openshift.io", Version: "v1beta1"}
```

Comment:
Thoughts on aligning the group to what AWS is doing?
https://github.com/openshift/cluster-api-provider-aws/blob/master/pkg/apis/awsprovider/v1beta1/register.go#L40

So, this would become "ibmcloudproviderconfig.openshift.io"

Reply:
And would require changes elsewhere.

Reply:
Well it seems there isn't consistency here. GCP has it like you've got it. https://github.com/openshift/cluster-api-provider-gcp/blob/master/pkg/apis/gcpprovider/v1beta1/register.go#L22

Reply:
+1, i think you will want this for the openshift version

Reply:
fwiw, azure also uses the `providerconfig` form. i would probably use the more explicit name for clarity.

https://github.com/openshift/cluster-api-provider-azure/blob/master/pkg/apis/azureprovider/v1beta1/register.go#L37